File: marinetes/user-mobile/app/src/main/kotlin/com/marinetes/network/serializations/marinetes/user/api/Diarist.kt
package com.marinetes.network.serializations.marinetes.user.api

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class Diarist(
  val id: String,
  @SerialName("full_name")
  val fullName: String,
  val birthdate: String,
  val document: String,
  @SerialName("general_register")
  val generalRegister: String,
  val phone: String,
  val email: String,
  @SerialName("accepting_services")
  val acceptingServices: Boolean,
  val avatar: String? = null,
  @SerialName("created_at")
  val createdAt: String,
  @SerialName("updated_at")
  val updatedAt: String,
)

File: marinetes/monorepo-legacy/shared/types/dtos/user/api/data/diarist.ts
import type { Diarist, LeanModel } from '@marinetes/types/model';

type PreRegisterFields =
  | 'full_name'
  | 'birthdate'
  | 'email'
  | 'document'
  | 'general_register';

// Create

export type DiaristCreateData = Pick<LeanModel<Diarist>, PreRegisterFields>;
